lib/main.dart

```diff
@@ -7,109 +9,225 @@ void main() {
+                  child: SizedBox(
+                    child: Center(
+                      child: Text(
+                        'Дані оновлено ${DateFormat('dd.MM.yyyy о HH:mm').format(DateTime.now())}',
+                        style: const TextStyle(
+                          fontSize: 14,
+                          color: Colors.white,
+                        ),
+                      ),
+                    ),
+                  ),
+                ),
+                Expanded(
+                  child: Container(
+                    padding: size.height >= size.width
+                        ? EdgeInsets.symmetric(horizontal: size.width / 2.5)
+                        : EdgeInsets.only(
+                            left:
+                                size.height / (size.height / size.width * 2.25),
+                            right:
+                                size.height / (size.height / size.width * 2.25),
+                          ),
+                    child: GridView.builder(
+                      itemCount: cards.length,
+                      itemBuilder: (_, i) => Container(
+                        decoration: BoxDecoration(
+                            color:
+                                i == _index ? Colors.black : cardColors[_index],
+                            shape: BoxShape.circle),
+                      ),
+                      gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
+                        crossAxisCount: cards.length,
+                        crossAxisSpacing: 5,
+                      ),
+                    ),
+                  ),
+                ),
+              ],
             ),
-          ],
-        ),
+          ),
+        ],
+      ),
+      bottomNavigationBar: BottomNavigationBar(
+        type: BottomNavigationBarType.fixed,
+        backgroundColor: Colors.transparent,
+        elevation: 0,
+        iconSize: 30,
+        items: const [
+          BottomNavigationBarItem(
+            icon: Icon(Icons.insert_drive_file_rounded),
+            label: 'Документи',
+          ),
+          BottomNavigationBarItem(
+            icon: Icon(Icons.flash_on_rounded),
+            label: 'Послуги',
+          ),
+          BottomNavigationBarItem(
+            icon: Icon(Icons.notifications_outlined),
+            label: 'Повідомлення',
+          ),
+          BottomNavigationBarItem(
+            icon: Icon(Icons.menu_rounded),
+            label: 'Меню',
+          ),
+        ],
+        selectedLabelStyle: const TextStyle(fontSize: 12),
+        unselectedLabelStyle: const TextStyle(fontSize: 12),
+        selectedItemColor: Colors.white,
+        unselectedItemColor: Colors.white,
       ),
-      floatingActionButton: FloatingActionButton(
-        onPressed: _incrementCounter,
-        tooltip: 'Increment',
-        child: const Icon(Icons.add),
-      ), // This trailing comma makes auto-formatting nicer for build methods.
     );
   }
 }

```

Comment:
It`s all is very pretty nice! 
but maybe, I will divide all this code into components for more comfortable work with a lot of line code